### ISSUE TITLE
Logs to ElasticSearch missing full information from version 23.0.0

### ISSUE THREAD

**Issue by hubertciebiada** [thepirat000/Audit.NET#660]
**Describe the bug**
When I upgrade Audit.NET.Elasticsearch from 22.1.0 to 23.0.0 my logs to ElasticSearch are missing entire Actions field in JSON sent to ES:

22.1.0
```
{
  "_index": "X",
  "_type": "_doc",
  "_id": "864b3842-11fb-4de2-9abc-c6c9d45f8837",
  "_version": 1,
  "_score": null,
  "_source": {
    "action": {
      "traceId": "0HN2AFHEJ2Q75:00000005",
      "httpMethod": "GET",
      "requestUrl": "https://localhost:7018/weatherforecast",
      "ipAddress": "::1",
      "responseStatus": "OK",
      "responseStatusCode": 200,
      "requestBody": {},
      "responseBody": {
        "type": "application/json; charset=utf-8",
        "value": "[{\"date\":\"2024-03-23\",\"temperatureC\":31,\"summary\":\"Cool\",\"temperatureF\":87},{\"date\":\"2024-03-24\",\"temperatureC\":18,\"summary\":\"Mild\",\"temperatureF\":64},{\"date\":\"2024-03-25\",\"temperatureC\":16,\"summary\":\"Hot\",\"temperatureF\":60},{\"date\":\"2024-03-26\",\"temperatureC\":-1,\"summary\":\"Mild\",\"temperatureF\":31},{\"date\":\"2024-03-27\",\"temperatureC\":36,\"summary\":\"Bracing\",\"temperatureF\":96}]"
      },
      "responseHeaders": {
        "content-Type": "application/json; charset=utf-8"
      },
      "customFields": {}
    },
    "eventType": "GET https://localhost:7018/weatherforecast",
    "environment": {
      "userName": "X",
      "machineName": "X",
      "domainName": "X",
      "callingMethodName": "System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start()",
      "assemblyName": "System.Private.CoreLib, Version=8.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e",
      "culture": "pl-PL",
      "customFields": {}
    },
    "customFields": {},
    "startDate": "2024-03-22T14:28:42.2336315Z",
    "endDate": "2024-03-22T14:28:42.2711031Z",
    "duration": 37
  },
  "fields": {
    "endDate": [
      "2024-03-22T14:28:42.271Z"
    ],
    "startDate": [
      "2024-03-22T14:28:42.233Z"
    ]
  },
  "sort": [
    1711117722233
  ]
}
```

23.0.0
```{
  "_index": "X",
  "_type": "_doc",
  "_id": "4f9266b4-fd73-479f-92e7-09d8c5c61a30",
  "_version": 1,
  "_score": null,
  "_source": {
    "eventType": "GET https://localhost:7018/weatherforecast",
    "environment": {
      "userName": "X",
      "machineName": "X",
      "domainName": "X",
      "callingMethodName": "System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start()",
      "assemblyName": "System.Private.CoreLib, Version=8.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e",
      "culture": "pl-PL",
      "customFields": {}
    },
    "customFields": {},
    "startDate": "2024-03-22T14:39:12.5484810Z",
    "endDate": "2024-03-22T14:39:12.6367951Z",
    "duration": 88
  },
  "fields": {
    "endDate": [
      "2024-03-22T14:39:12.636Z"
    ],
    "startDate": [
      "2024-03-22T14:39:12.548Z"
    ]
  },
  "sort": [
    1711118352548
  ]
}
```

**To Reproduce**
I've created a repo that mimics our config. If you switch from 22.1.0 to 23.0.0 the behavior should be replicable. You will have to provide your own ES address and index.
https://github.com/hubertciebiada/auditnetissue

**Expected behavior**
I expect actions field to be saved to the ES. It may be saved with other name, but I need the values. :)

**Libraries (specify the Audit.NET extensions being used including version):**
```
   <PropertyGroup>
        <TargetFramework>net8.0</TargetFramework>
        <Nullable>enable</Nullable>
        <ImplicitUsings>enable</ImplicitUsings>
    </PropertyGroup>

    <ItemGroup>
        <PackageReference Include="Audit.NET.Elasticsearch" Version="23.0.0" />
        <PackageReference Include="Audit.WebApi.Core" Version="25.0.3" />
        <PackageReference Include="Microsoft.AspNetCore.OpenApi" Version="8.0.3" />
        <PackageReference Include="Swashbuckle.AspNetCore" Version="6.5.0" />
    </ItemGroup>
```

**Target .NET framework:**
 - .NET 8.0

Let me know if there is anything more I can provide. :)

Great library BTW!

**Comment by thepirat000** [comment]
The problem is related to System.Text.Json serializer. Before version 22.1 this library was using NewtonsoftJson, but changed to use System.Text.Json.

You can try adding a reference to the package `NEST.JsonNetSerializer` and configure the audit data provider as follows:

```c#
using Nest.JsonNetSerializer;

var pool = new SingleNodeConnectionPool(new Uri("my ES addres"));
var connectionSettings = new ConnectionSettings(pool, sourceSerializer: JsonNetSerializer.Default);

Configuration.Setup()
    .UseElasticsearch(config => config
        .ConnectionSettings(connectionSettings)
        .Index(_ => "my_index")
        .Id(_ => Guid.NewGuid()));
```

I will take a deeper look this weekend, maybe that should be the default behavior

**Comment by thepirat000** [comment]
This was fixed in version 25.0.4, please upgrade your references and re-test. It should work with both the default serializer and the JsonNetSerializer

**Comment by hubertciebiada** [comment]
I've only checked with the default serializer, but it works. Thank you very much! :)
